pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -298,7 +309,12 @@ func NewNamedMapFromBuildPlan(build *model.Build) (artifact.NamedMap, error) {
 // adds the artifact's dependencies to the map. This is useful for when we are
 // using the BuildLogStreamer as it operates on the older recipeID and will include
 // more artifacts than what we originally calculated in the runtime closure.
-func AddBuildArtifacts(artifactMap artifact.Map, build *model.Build) error {
+func AddBuildArtifacts(artifactMap artifact.Map, build *model.Build) (artifact.Map, error) {
```

Comment:
You've removed the only call to this function in `setup.go`. Do we even need this anymore?